src/FrequencyMapAgent.cpp

```diff
@@ -441,5 +492,21 @@ namespace geopm
         m_core_freq_max = m_platform_io.read_signal("CPU_FREQUENCY_MAX_AVAIL", GEOPM_DOMAIN_BOARD, 0);
         m_uncore_init_min = m_platform_io.read_signal("CPU_UNCORE_FREQUENCY_MIN_CONTROL", GEOPM_DOMAIN_BOARD, 0);
         m_uncore_init_max = m_platform_io.read_signal("CPU_UNCORE_FREQUENCY_MAX_CONTROL", GEOPM_DOMAIN_BOARD, 0);
+        m_gpu_freq_ctl_domain_type = m_platform_io.control_domain_type("GPU_CORE_FREQUENCY_MAX_CONTROL");
+
+        if (m_gpu_freq_ctl_domain_type == GEOPM_DOMAIN_INVALID) {
+            m_num_gpu_freq_ctl_domain = 0;
+        }
+        else {
+            m_num_gpu_freq_ctl_domain = m_platform_topo.num_domain(m_gpu_freq_ctl_domain_type); 
+            m_gpu_min = m_platform_io.read_signal("GPU_CORE_FREQUENCY_MIN_AVAIL", m_gpu_freq_ctl_domain_type, 0);
+            m_gpu_max = m_platform_io.read_signal("GPU_CORE_FREQUENCY_MAX_AVAIL", m_gpu_freq_ctl_domain_type, 0);
+
+            for (size_t ctl_idx = 0; ctl_idx < (size_t) m_num_gpu_freq_ctl_domain; ++ctl_idx) {
+                m_gpu_control_idx.push_back(m_platform_io.push_control("GPU_CORE_FREQUENCY_MAX_CONTROL",
+                                                                       m_gpu_freq_ctl_domain_type,
+                                                                       ctl_idx));
```

Comment:
```suggestion
                m_gpu_min_control_idx.push_back(m_platform_io.push_control("GPU_CORE_FREQUENCY_MIN_CONTROL",
                                                                       m_gpu_freq_ctl_domain_type,
                                                                       ctl_idx));
                m_gpu_max_control_idx.push_back(m_platform_io.push_control("GPU_CORE_FREQUENCY_MAX_CONTROL",
                                                                       m_gpu_freq_ctl_domain_type,
                                                                       ctl_idx));
```

Setting only the maximum frequency allows the driver & fw to lower frequency below it in certain conditions, which may lead to noisier data overall (ex: V100s will swing between 135MHz at idle to 1290Mhz - Max while active).  

In addition for any system that imposes a strict min <= max requirement only setting the maximum frequency will cause an agent to fail.  In order to address this both the [NVML](https://github.com/geopm/geopm/blob/dev/service/src/NVMLIOGroup.cpp#L540) & [LevelZero](https://github.com/geopm/geopm/blob/dev/service/src/LevelZeroIOGroup.cpp#L914) IOGroups support control write re-ordering on failure if both the maximum & minimum setting are being written in batch.